engine/core/src/main/java/org/datacleaner/job/builder/AbstractComponentBuilder.java

```diff
@@ -472,6 +472,18 @@ private void synchronizeDependentProperties(final ConfiguredPropertyDescriptor p
         }
     }
 
+    private static String getKey(final Object object) {
+        if (object instanceof InputColumn<?>) {
+            final InputColumn<?> inputColumn = (InputColumn<?>) object;
+
+            if (inputColumn.isVirtualColumn()) {
+                return inputColumn.getName();
```

Comment:
Ah, but that's the CSV file writer that won't produce a file with non-unique columns. Nothing to do with the job. If the job had disallowed it, you wouldn't have had two columns with the same name in the first place.